.github/workflows/build.yml

```diff
@@ -0,0 +1,58 @@
+name: CI
+
+on:
+  workflow_dispatch:
+    inputs:
+      keep_workdir:
+        required: false
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  check:
+    if: |
+      github.event.issue.pull_request &&
+      startsWith(github.event.comment.body, '/ost') &&
+      (
+        github.event.comment.author_association == 'MEMBER' ||
+        github.event.comment.author_association == 'COLLABORATOR'
+      )
+
+  build-el8stream:
+    needs: check
+    runs-on: image-builders
+    container:
+      image: quay.io/ovirt/buildcontainer:el9stream
+      options: --group-add qemu --user build --privileged
+
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: build
+      run: |
+        cd ..
+        curl --fail -O https://resources.ovirt.org/repos/ovirt/github-ci/centos/CentOS-Stream.iso
+        DISTRO=el8stream ost-images/build.sh
+
+    - name: Upload rpm to resources.ovirt.org
+      uses: ovirt/ovirt-resources-upload-action@main
+      with:
+        username: ${{ secrets.SSH_USERNAME_FOR_RESOURCES_OVIRT_ORG }}
+        key: ${{ secrets.SSH_KEY_FOR_RESOURCES_OVIRT_ORG }}
+        known_hosts: ${{ secrets.KNOWN_HOSTS_FOR_RESOURCES_OVIRT_ORG }}
+        source: rpmbuild/RPMS/x86_64/*.rpm
+        target: github-ci/ost-images
+        cleanup: yes
+        createrepo: yes
+        # keep 10 last rpms (not necessarily complete set of images) + repodata
+        keep_files_count: 11
+
+    - name: Clean up entire workdir
+      if: always()
```

Comment:
That construct provides something like a `finally` block?

Reply:
yes
it's jsut that we're a bit short on space if all the projects left their images and squashfs and whatnot...